pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -52,11 +52,35 @@ const (
 	deleteClusterconfigCommitMessage  = "Delete commit of cluster configuration; generated by EKS-A CLI"
 )
 
+// GitClient is responsible for provider-agnostic actions on a generic git repository.
+// Any actions that would be take with a git CLI executable should be included in the GitClient
+type GitClient interface {
+	Add(filename string) error
+	Remove(filename string) error
+	Clone(ctx context.Context) error
+	Commit(message string) error
+	Push(ctx context.Context) error
+	Pull(ctx context.Context, branch string) error
+	Init() error
+	Branch(name string) error
+}
+
+// GitProviderClient is responsible for provider-dependant actions on a hosted git provider API
+// Any actions you would take via the API of a git hosting provider should be included in the GitProviderClient
+type GitProviderClient interface {
```

Comment:
Based on the discussion above, could this have a better name? `GitRepositoryAdministrator` for example.

Reply:
I can take a look at a more narrow name; this is going to be changing again today so will think about it